BrowserKit/Sources/Redux/Store.swift

```diff
@@ -68,15 +70,31 @@ public class Store<State: StateType>: DefaultDispatchStore {
     }
 
     public func dispatch(_ action: Action) {
-        guard Thread.isMainThread && !actionRunning else {
-            DispatchQueue.main.async { [weak self] in self?.dispatch(action) }
-            return
-        }
-
         logger.log("Dispatched action: \(action.displayString())", level: .info, category: .redux)
 
-        actionRunning = true; defer { actionRunning = false }
+        if Thread.isMainThread {
+            actionQueue.append(action)
+            processQueuedActions()
+        } else {
+            DispatchQueue.main.async { [weak self] in
+                self?.actionQueue.append(action)
+                self?.processQueuedActions()
```

Comment:
I believe this should just re-dispatch the action on the main thread, and then the code above will append and process the action on the MT.

```suggestion
                self?.dispatch(action)
```